

fs/reiser4/debug.c: In function `reiser4_print_prefix':
fs/reiser4/debug.c:144: warning: implicit declaration of function `in_interrupt'
fs/reiser4/debug.c:144: warning: implicit declaration of function `in_irq'

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/reiser4/context.c                  |    1 +
 25-akpm/fs/reiser4/debug.c                    |    1 +
 25-akpm/fs/reiser4/log.c                      |    1 +
 25-akpm/fs/reiser4/plugin/compress/compress.c |    1 +
 4 files changed, 4 insertions(+)

diff -puN fs/reiser4/debug.c~reiser4-debug-build-fix fs/reiser4/debug.c
--- 25/fs/reiser4/debug.c~reiser4-debug-build-fix	2004-09-15 22:10:18.219205248 -0700
+++ 25-akpm/fs/reiser4/debug.c	2004-09-15 22:10:18.229203728 -0700
@@ -44,6 +44,7 @@
 #include <linux/vmalloc.h>
 #include <linux/ctype.h>
 #include <linux/sysctl.h>
+#include <linux/hardirq.h>
 
 /*
  * global buffer where message given to reiser4_panic is formatted.
diff -puN fs/reiser4/context.c~reiser4-debug-build-fix fs/reiser4/context.c
--- 25/fs/reiser4/context.c~reiser4-debug-build-fix	2004-09-15 22:10:18.221204944 -0700
+++ 25-akpm/fs/reiser4/context.c	2004-09-15 22:10:18.229203728 -0700
@@ -38,6 +38,7 @@
 #include "context.h"
 
 #include <linux/writeback.h> /* balance_dirty_pages() */
+#include <linux/hardirq.h>
 
 #if REISER4_DEBUG_CONTEXTS
 /* List of all currently active contexts, used for debugging purposes.  */
diff -puN fs/reiser4/log.c~reiser4-debug-build-fix fs/reiser4/log.c
--- 25/fs/reiser4/log.c~reiser4-debug-build-fix	2004-09-15 22:10:18.222204792 -0700
+++ 25-akpm/fs/reiser4/log.c	2004-09-15 22:10:18.230203576 -0700
@@ -41,6 +41,7 @@
 #include <linux/slab.h>
 #include <linux/bio.h>
 #include <linux/vmalloc.h>
+#include <linux/hardirq.h>
 
 #if REISER4_LOG
 
diff -puN fs/reiser4/plugin/compress/compress.c~reiser4-debug-build-fix fs/reiser4/plugin/compress/compress.c
--- 25/fs/reiser4/plugin/compress/compress.c~reiser4-debug-build-fix	2004-09-15 22:10:18.224204488 -0700
+++ 25-akpm/fs/reiser4/plugin/compress/compress.c	2004-09-15 22:10:18.230203576 -0700
@@ -10,6 +10,7 @@
 #include <linux/zlib.h>
 #include <linux/vmalloc.h>
 #include <linux/types.h>
+#include <linux/hardirq.h>
 
 /******************************************************************************/
 /*                         null compression                                   */
_
